conftest.py

```diff
@@ -43,12 +44,26 @@
 }
 
 
+def wait_for_cache_updates(component_catalog):
+    """
+    TODO
+    """
+    t0 = time.time()
+    elapsed_time = t0 - time.time()
+    component_catalog.wait_for_all_manifest_updates()
+    while component_catalog.cache_manager.manifest_update_pending and elapsed_time < 60:
```

Comment:
Are you finding this `while` loop to be necessary?  I figured `join`ing the threads would be sufficient - just curious.

Reply:
Yeah the more I thought about this loop, the more I don't like it. The way things are currently set up, `join`ing doesn't work, but I think I'm going to shift around when various threads mark their tasks as `done` so that we can avoid this